### PR TITLE
Provide generic infrastructure for updating moddle properties

### PR DESCRIPTION
**Pull request by nikku**
This is a proposal to improve upon https://github.com/bpmn-io/bpmn-js/pull/1365.

#### What is inside

* Provided a generic API to update moddle elements (that may be backed up by one or more diagram elements)

  ```
  modeling.updateModdleProperties(element, moddleElement, properties);
  ```

* Clean up some wording: All handlers are suffixed with `Handler`
* Adjust API to feel like other APIs: We by convention pass the affected diagram element as the first argument (the element in which context the modeling operation executes)

----

Note to @azeghers: Please pay attention to https://github.com/bpmn-io/bpmn-js/commit/5383a44116700b634c2a38af095b5ba70a1e58f2. Prototypes must be declared outside a constructor function.

### CODE REVIEW

**Comment by philippfromme** [comment]
I think you can squash the first two commits.

**Comment by nikku** [comment]
Need to trick codecov first. Will squash then :cowboy_hat_face:

File: lib/features/modeling/cmd/UpdateModdlePropertiesHandler.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ *
+ * @param {ModdleElement} moddleElement
+ *
+ * @return {Array<djs.model.Element>}
+ */
+UpdateModdlePropertiesHandler.prototype.getVisualReferences = function(moddleElement) {
+
+  var elementRegistry = this._elementRegistry;
+
+  if (is(moddleElement, 'bpmn:DataObject')) {
```
Comment (philippfromme) [inline lib/features/modeling/cmd/UpdateModdlePropertiesHandler.js:61]: Do we know of any other elements that visually reflect a nested moddle object's state?

Comment (nikku) [reply lib/features/modeling/cmd/UpdateModdlePropertiesHandler.js:61]: Data stores, data inputs, data outputs and generally extension elements are candidates.

Comment (barmac) [reply lib/features/modeling/cmd/UpdateModdlePropertiesHandler.js:61]: I checked `bpmn:DataStore` and has only one representation regardless of its properties.

**Comment by nikku** [comment]
Tricked `codecov`, squashed dat thing.

**Review by barmac — APPROVED** [review]
LGTM

**Comment by barmac** [comment]
This is a breaking change, though. We are removing an existing Modeling method.

**Comment by nikku** [comment]
> This is a breaking change, though. We are removing an existing Modeling method.

Nope. We introduced that previous, unreleased API just a day ago.

**Comment by barmac** [comment]
Ok sorry 😅

**Pull request merged**